### DISCUSSION TITLE
Add a pull request template

### DISCUSSION THREAD

**Discussion by sdilts** [stumpwm/stumpwm#1119] [Ideas]
New pull requests are great to see, but they don't always contain all of the information needed to verify if the bugfix or new feature should be merged. I propose we add a pull request template that requests some basic information. Here's the documentation on how to do that: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

The goal here is to help make the review process smoother for both the contributors and reviewers, not add a lengthy checklist that contributors need to wade through before submitting their patch. Here's the information I'd like to request for each type of patch:

## Bugfixes
+ A link to an issue that describes the problem, or repeatable instructions on how to trigger the bug. Some bugs exist because the documentation is bad, something is being misused, or the feature itself is confusing, so this could also include an explaination on WHY this is a bug.
+ At least a one liner on how this fixes the issue, or why it existed in the first place.

## New Features
+ A detailed description on how to use the new feature. Documentation updates by themselves are fine, but if it's something bigger, a lengthier pull request description would be good too. Config code, screen captures, etc are great to include in these.
+ An explanation of why the new feature is needed or improves Stumpwm.
+ A request to review the feature and check if it would fit better in the contrib repo. (What are the criteria for the contrib vs main repo?)

We could also point to good PRs within our own repo as examples of what a good PR looks like.

**Comment by szos** [comment]
I think this would be very useful, and may make some folks more comfortable with opening PRs. What are some good PRs that could be pointed to?

**Comment by dmb2** [comment]
I like this idea!

**Comment by szos** [comment]
Building on this, I think having issue templates/guidelines would also be a good move.
